src/main/java/org/broadinstitute/consent/http/db/mapper/DraftSubmissionInterfaceMapper.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import java.lang.reflect.InvocationTargetException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.UUID;
+import org.broadinstitute.consent.http.exceptions.NoMatchingClassException;
+import org.broadinstitute.consent.http.models.DraftInterface;
+import org.broadinstitute.consent.http.models.User;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+
+public class DraftSubmissionInterfaceMapper implements RowMapper<DraftInterface>, RowMapperHelper {
+
+  @Override
+  public DraftInterface map(ResultSet rs, StatementContext ctx) throws SQLException, NoMatchingClassException {
+    if (!hasColumn(rs, "schema_class")) {
+      throw new NoMatchingClassException("Missing class name.");
+    }
+
+    DraftInterface dsi;
+    String className = rs.getString("schema_class");
+
+    try {
+      dsi = (DraftInterface) Class.forName(className).getDeclaredConstructor().newInstance();
```

Comment:
I think that allowing arbitrary class instantiation here is unnecessarily general. Unless we end up supporting a system where new draft types can be added via linking a .jar to the project, adding a new draft requires a code change, and the list of known draft types are known at compile time (vs link time).

One way to write this without general support is to add a static `Map` that maps a draft name to its constructor. The map could be initialized either directly in the class that declares it, or each draft type could have a static initializer that adds an entry to the static map. Instead of a `Map` an enum could be used, as enums are also static maps of keys to a set of properties.

Reply:
I've refactored this a bit, based on a quick chat I had with Greg.  It removes the reflection and adds a DraftType enum with a small utility to return the proper implementation based on the Draft Type.